### ISSUE TITLE
Saving new passwords doesn't work again (with postgres-db)

### ISSUE THREAD

**Issue by wollomatic** [marius-wieschollek/passwords#417]
### System Information
**Server:**
```
{
    "version": {
        "server": "21.0.3.1",
        "app": "2021.7.23",
        "lsr": false,
        "php": "7.4.21"
    },
    "environment": {
        "os": "Linux",
        "architecture": "x86_64",
        "bits": 64,
        "database": "pgsql",
        "cron": "cron",
        "proxy": false,
        "sslProxy": true,
        "subdirectory": false
    },
    "legacyApi": {
        "enabled": 0,
        "used": false
    },
    "services": {
        "images": "imagick",
        "favicons": "bi",
        "previews": "default",
        "security": "hibp",
        "words": "random",
        "previewApi": false,
        "faviconApi": false
    },
    "settings": {
        "channel": "stable",
        "nightlies": false,
        "handbook": false,
        "performance": 5
    },
    "encryption": {
        "sse": {
            "SSEv1r1": false,
            "SSEv1r2": true,
            "SSEv2r1": true,
            "none": true,
            "default": "SSEv1r2"
        },
        "cse": {
            "CSEv1r1": true,
            "none": true,
            "default": "none"
        }
    }
}

 Database and Version: postgres (PostgreSQL) 12.7 (Debian 12.7-1.pgdg100+1)

 Nextcloud and database are docker images: nextcloud:21 / postgres:12
```

**Client:**
```
 Browser and Version: Firefox 90.0.2 (64-Bit)
 Client OS and Version: Windows 10 21H1
```

### Steps to reproduce
Click on "+ / Neues Passwort" to add new password, enter data and save with "Speichern"

### Expected result
New password should be saved

### Actual result
Two messages: "Erstellen des Passworts Fehlgeschlagen" / "Resource not found"

### Nextcloud log
<details>
<summary>Nextcloud log</summary>
{"reqId":"YOAuNQ6GBmGh7guXyZsJ","level":3,"time":"2021-08-05T19:25:44+00:00","remoteAddr":"[removed]","user":"[removed}","app":"passwords","method":"POST","url":"/index.php/apps/passwords/api/1.0/password/create","message":"Error \"Did expect one result but found none when executing: query \"SELECT * FROM `*PREFIX*passwords_password` WHERE (`deleted` = :dcValue1) AND (`user_id` = :dcValue2) AND (`uuid` = :dcValue3)\"; \" in OCA\\Passwords\\Controller\\Api\\PasswordApiController::create","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:90.0) Gecko/20100101 Firefox/90.0","version":"21.0.3.1"}
{"reqId":"YOAuNQ6GBmGh7guXyZsJ","level":4,"time":"2021-08-05T19:25:44+00:00","remoteAddr":"[removed]","user":"[removed]","app":"passwords","method":"POST","url":"/index.php/apps/passwords/api/1.0/password/create","message":{"Exception":"OCP\\AppFramework\\Db\\DoesNotExistException","Message":"Did expect one result but found none when executing: query \"SELECT * FROM `*PREFIX*passwords_password` WHERE (`deleted` = :dcValue1) AND (`user_id` = :dcValue2) AND (`uuid` = :dcValue3)\"; ","Code":0,"Trace":[{"file":"/var/www/html/lib/public/AppFramework/Db/QBMapper.php","line":359,"function":"findOneQuery","class":"OCP\\AppFramework\\Db\\QBMapper","type":"->","args":[{"__class__":"OC\\DB\\QueryBuilder\\QueryBuilder"}]},{"file":"/var/www/html/custom_apps/passwords/lib/Db/AbstractMapper.php","line":201,"function":"findEntity","class":"OCP\\AppFramework\\Db\\QBMapper","type":"->","args":[{"__class__":"OC\\DB\\QueryBuilder\\QueryBuilder"}]},{"file":"/var/www/html/custom_apps/passwords/lib/Db/AbstractMapper.php","line":174,"function":"findOneByFields","class":"OCA\\Passwords\\Db\\AbstractMapper","type":"->","args":[["uuid","b1b40a2d-42f1-47f5-8c5e-553f0d987292",2,"eq"]]},{"file":"/var/www/html/custom_apps/passwords/lib/Db/AbstractMapper.php","line":99,"function":"findOneByField","class":"OCA\\Passwords\\Db\\AbstractMapper","type":"->","args":["uuid","b1b40a2d-42f1-47f5-8c5e-553f0d987292"]},{"file":"/var/www/html/custom_apps/passwords/lib/Services/Object/AbstractModelService.php","line":58,"function":"findByUuid","class":"OCA\\Passwords\\Db\\AbstractMapper","type":"->","args":["b1b40a2d-42f1-47f5-8c5e-553f0d987292"]},{"file":"/var/www/html/custom_apps/passwords/lib/Services/ValidationService.php","line":250,"function":"findByUuid","class":"OCA\\Passwords\\Services\\Object\\AbstractModelService","type":"->","args":["b1b40a2d-42f1-47f5-8c5e-553f0d987292"]},{"file":"/var/www/html/custom_apps/passwords/lib/Services/ValidationService.php","line":59,"function":"checkHash","class":"OCA\\Passwords\\Services\\ValidationService","type":"->","args":[{"id":null,"__class__":"OCA\\Passwords\\Db\\PasswordRevision"}]},{"file":"/var/www/html/custom_apps/passwords/lib/Services/Object/PasswordRevisionService.php","line":124,"function":"validatePassword","class":"OCA\\Passwords\\Services\\ValidationService","type":"->","args":[{"id":null,"__class__":"OCA\\Passwords\\Db\\PasswordRevision"}]},{"file":"/var/www/html/custom_apps/passwords/lib/Controller/Api/PasswordApiController.php","line":148,"function":"create","class":"OCA\\Passwords\\Services\\Object\\PasswordRevisionService","type":"->","args":["b1b40a2d-42f1-47f5-8c5e-553f0d987292","d4ddad598952f9ec7a7ba2bd10a7b5083fd291d8d9458aaa902150cf8f56236f34265c72f1f40df60ca5ed50","68f802ed4bd4b6a8f52d6d338aeaedfb2d550d366709e5fae389a91db034cbb7ece5549491b186af765d0932c8","f080634f-0721-45ac-9964-5d947bce0c0c","CSEv1r1","e416dc30043f9b10e817d2c5b3256e","eca630c60ec438ce2449ca0a47962a86837f85df371138f7f74fbbc84440b830da7a5f87bfee52dc0835b786","","","aad8f66e790463f9505bd3333f867d316e727e0a5f677bc037aedd281793cfd569a2e5f7c4a96c3a08d4","17d3192f-5248-41bc-99a8-d92aad5e6883",1628191544,false,false,false]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":218,"function":"create","class":"OCA\\Passwords\\Controller\\Api\\PasswordApiController","type":"->","args":["d4ddad598952f9ec7a7ba2bd10a7b5083fd291d8d9458aaa902150cf8f56236f34265c72f1f40df60ca5ed50","68f802ed4bd4b6a8f52d6d338aeaedfb2d550d366709e5fae389a91db034cbb7ece5549491b186af765d0932c8","f080634f-0721-45ac-9964-5d947bce0c0c","CSEv1r1","e416dc30043f9b10e817d2c5b3256e","eca630c60ec438ce2449ca0a47962a86837f85df371138f7f74fbbc84440b830da7a5f87bfee52dc0835b786","","","aad8f66e790463f9505bd3333f867d316e727e0a5f677bc037aedd281793cfd569a2e5f7c4a96c3a08d4","17d3192f-5248-41bc-99a8-d92aad5e6883",1628191544,false,false,[]]},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":127,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Passwords\\Controller\\Api\\PasswordApiController"},"create"]},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Passwords\\Controller\\Api\\PasswordApiController"},"create"]},{"file":"/var/www/html/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["PasswordApiController","create",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"passwords.password_api.create"}]},{"file":"/var/www/html/lib/base.php","line":993,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/passwords/api/1.0/password/create"]},{"file":"/var/www/html/index.php","line":37,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/html/lib/public/AppFramework/Db/QBMapper.php","Line":281,"CustomMessage":"Did expect one result but found none when executing: query \"SELECT * FROM `*PREFIX*passwords_password` WHERE (`deleted` = :dcValue1) AND (`user_id` = :dcValue2) AND (`uuid` = :dcValue3)\"; "},"userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:90.0) Gecko/20100101 Firefox/90.0","version":"21.0.3.1"}
</details>

### Browser log
<details>
<summary>Browser log</summary>
XHRPOSThttps://[removed]/index.php/apps/passwords/api/1.0/password/create
[HTTP/2 404 Not Found 526ms]

Object { status: "error", id: 404, message: "Resource not found", response: Response }
​
id: 404
​
message: "Resource not found"
​
response: Response { type: "basic", url: "https://[removed]/index.php/apps/passwords/api/1.0/password/create", redirected: false, … }
​
status: "error"
​
<prototype>: Object { … }
app.js:45:188284
    _apiErrorEvent https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:45
    _registerApiErrorEvent https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:45
    emit https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _processJsonResponse https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:45
    _sendRequest https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:45
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:45
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:45
    createPasswordFromData https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    createPasswordFromData https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    _success https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    submitAction https://[removed]/custom_apps/passwords/js/Static/CreatePassword.f71d4ab09ff1d475b489.js:10
    submit https://[removed]/custom_apps/passwords/js/Static/CreatePassword.f71d4ab09ff1d475b489.js:1
    Jt https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    r https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _wrapper https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    (Async: EventListener.handleEvent)
    $n https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    le https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    ei https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    m https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    A https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    A https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    A https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Oi https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _update https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    n https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    get https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Ar https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    init https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Oi https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _update https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    n https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    get https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Ar https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    click https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Jt https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    r https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _wrapper https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
Object { status: "error", id: 404, message: "Resource not found", response: Response }
​
id: 404
​
message: "Resource not found"
​
response: Response { type: "basic", url: "https://[removed]/index.php/apps/passwords/api/1.0/password/create", redirected: false, … }
​
status: "error"
​
<prototype>: Object { … }
app.js:25:94844
    createPasswordFromData https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    (Async: promise callback)
    createPasswordFromData https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    createPasswordFromData https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    _success https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    submitAction https://[removed]/custom_apps/passwords/js/Static/CreatePassword.f71d4ab09ff1d475b489.js:10
    submit https://[removed]/custom_apps/passwords/js/Static/CreatePassword.f71d4ab09ff1d475b489.js:1
    Jt https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    r https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _wrapper https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    (Async: EventListener.handleEvent)
    $n https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    le https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    ei https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    m https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    A https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    A https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    A https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Oi https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _update https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    n https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    get https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Ar https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    init https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    c https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Oi https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _update https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    n https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    get https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Ar https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    $mount https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:25
    createPassword https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    click https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    Jt https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    r https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
    _wrapper https://[removed]/custom_apps/passwords/js/Static/app.js?v=fc5695fa-15:43
Uncaught (in promise) 
Object { id: null, username: "[removed]", password: "[removed]", label: "[removed]", url: null, notes: "", customFields: "[]", folder: "17d3192f-5248-41bc-99a8-d92aad5e6883", edited: 0, hidden: false, … }
​
_encrypted: false
​
cseKey: ""
​
cseType: "CSEv1r1"
​
customFields: "[]"
​
edited: 0
​
favorite: false
​
folder: "17d3192f-5248-41bc-99a8-d92aad5e6883"
​
hidden: false
​
id: null
​
label: "[removed]"
​
notes: ""
​
password: "[removed]"
​
tags: Array []
​
trashed: false
​
url: null
​
username: "[removed]"
​
<prototype>: Object { … }</details>

**Comment by 9hawk8** [comment]
Same for me here 💯

But this is only an issue within the Browser opening up the Nextcloud Account.
Saving works by adding through Apps or the Firefox-Addon.

Not experimented with it so far or even analyzed the logs, but just to mention it, I have both, CSE **and** SSE ("Erweiterte Verschlüsselung"), active in the Passwords settings.

~~As @wollomatic said "not working again" it's maybe something handled before and broken by an Update?
Edit: This issue #397 seems  #related.~~



**System Information**

Server:
Nextcloud Version: 21.0.4
PHP Version: 7.4.9
Database and Version: MDB 10.3.27
Server OS and Version: DSM7.0
Passwords Version: 2021.7.23

Client:

Browser and Version: Firefox 91.0.1
Client OS and Version: Windows 10 21H1

**Comment by wollomatic** [comment]
Yes really: **This is only a problem when using passwords with libsodium enabled.**

This issue exists on all my instances, regardless of using MariaDB or PostgreSQL, so I doubt it's related to #397.

**Comment by 9hawk8** [comment]
Okay...
Sorry for confusing here than!

Just want to mention briefly: Just upgraded to the App-Version 2021.9.20.

Can confirm that actually now, by deactivating libsodium (correspondingly: selecting "On the server" for Encryption on the bottom left in the Create-Password-Window) it works. So this is as you said just a problem if one tries to create a new password on the browser using libsodium.

**Comment by SEsternO** [comment]
Can confirm this, with one more detail. Just installed 2021.10.20 version of app. Problem comes up if I use libsodium encryption for new passwords, however only when storing 50% or 75% of security check hash, while with no or full stored hash everything seems to work just fine.

Server: NC 22.2.0; MariaDB 10.5.11; nginx 1.18.0; PHP 7.4.23; Debian 11.
Client: Firefox 92.0.1.

**Comment by 9hawk8** [comment]
Just upgraded to App-Version 2021.10.20 too and NC-Version is now 21.0.5.

Indeed changing the option for how much of the security hash is saved to "save whole hash" it also works for me.
So there seems to be a problem with not storing the complete hash and using libsodium.

As you, @wollomatic, also can get round with disabling libsodium I am sure you also get it working by storing the whole security hash, like @SEsternO found out?

But as mentioned before mobile Apps and the Firefox-Extension/Add-On are all working great with all the options enabled, for me at least. It is just not working via the actual Browserinterface logged into your Nextcloud and opening up Passwords there.

Maybe @wollomatic, do you want to change the issues title to something more describing? As @SEsternO and me both having MDB up and running it is no more a problem only for postgres.
I'd suggest something like "Saving passwords does not work storing partial hash AND using libsodium"?

**Comment by wollomatic** [comment]
Problem solved in 2021.11.20. Thank you @marius-wieschollek
